facade-charm/tests/unit/test_charm.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2024 pietro
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+import unittest
+
+import ops
+import ops.testing
+from charm import FacadeCharmCharm
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self):
+        self.harness = ops.testing.Harness(FacadeCharmCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    def test_httpbin_pebble_ready(self):
+        # Expected plan after Pebble ready with default config
+        expected_plan = {
+            "services": {
+                "httpbin": {
+                    "override": "replace",
+                    "summary": "httpbin",
+                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
+                    "startup": "enabled",
+                    "environment": {"GUNICORN_CMD_ARGS": "--log-level info"},
+                }
+            },
+        }
+        # Simulate the container coming up and emission of pebble-ready event
+        self.harness.container_pebble_ready("httpbin")
+        # Get the plan now we've run PebbleReady
+        updated_plan = self.harness.get_container_pebble_plan("httpbin").to_dict()
+        # Check we've got the plan we expected
+        self.assertEqual(expected_plan, updated_plan)
+        # Check the service was started
+        service = self.harness.model.unit.get_container("httpbin").get_service("httpbin")
+        self.assertTrue(service.is_running())
+        # Ensure we set an ActiveStatus with no message
+        self.assertEqual(self.harness.model.unit.status, ops.ActiveStatus())
```

Comment:
What would you say about using `pytest` instead?

Reply:
+1, but again this is the standard `charmcraft init` profile.